juniper-vpn.py

```diff
@@ -42,7 +41,7 @@
 """
 
 def truncated_value(h):
-    bytes = map(ord, h)
+    bytes = list(map(ord, h))
```

Comment:
I've merged the backwards-compatible parts of this commit.

I'm sort of thinking it would be nice to do all the preparatory patches first, then flip the switch to Python 3 at the end. That way individual commits within the series will still work for everyone.

That would include the setup.py stuff - I guess we can't break Python 2 support without that, because installing via `pip` is going to be the easiest way for most people to get `mechanize` on Python 3.

Reply:
Writing code that works for both Python 2 and Python 3 is generically speaking a mess and requires the use of packages like `future` or `six`. Unless there is a compelling reason to do so, it's a much, much better idea to avoid getting into that. The situation usually is:
1) Python 2 is dead. Some distributions don't ship python2 by default. All distributions without exceptions that I know ship Python 3, and more specifically >= 3.4
2) the only really good reason to maintain Python 2 compatibility is to be able to mix with old/unmaintained code

Point 1) means that *all* users will have access to Python 3. At the very worst, it will require an `apt install python3`. Point 2) means we don't care about Python2, since this project is not a lib anyone else is trying to import, and this package does not rely on any 3rd party library that is Python 2 only.

Reply:
> this package does not rely on any 3rd party library that is Python 2 only.

This is only partially true, due to the mechanize dependency. `python3-mechanize` hasn't yet hit Debian stable. It can be installed from Pip, but so far I have managed to get away with having zero locally-installed Python packages... hence my reluctance to start now :)

This is also why I appreciate adding the wrapper scripts :) Although I reckon I might end up deleting the warnings...

Re. a "flag day" switch - having thought about it, it probably is OK. I was worried about people suddenly not being able to get into the VPN, but that would only happen if they've just updated this repo for some reason, in which case it'll be pretty obvious what's gone wrong.